test/jdk/com/sun/jndi/ldap/DeadSSLLdapTimeoutTest.java

```diff
@@ -141,50 +142,59 @@ public Boolean call() {
 
 class DeadSSLServer extends Thread {
     ServerSocket serverSock;
-    boolean accepting = false;
-
-    public DeadSSLServer() throws IOException {
-        this.serverSock = new ServerSocket(0);
+    // Latch to be used by client to wait for server to start
+    CountDownLatch serverStarted = new CountDownLatch(1);
+
+    // Latch to be used by server thread to wait for client to finish testing
+    CountDownLatch testDone = new CountDownLatch(1);
+
+    public DeadSSLServer(SocketAddress socketAddress) throws IOException {
+        // create unbound server socket
+        var srvSock = new ServerSocket();
+        // bind it to the address provided
+        srvSock.bind(socketAddress);
+        this.serverSock = srvSock;
         start();
     }
 
     public void run() {
-        while(true) {
-            try {
-                accepting = true;
-                Socket socket = serverSock.accept();
-            } catch (Exception e) {
-                break;
-            }
+        // Signal client to proceed with the test
+        serverStarted.countDown();
+        try (var socket = serverSock.accept()) {
+            System.err.println("Accepted connection:" + socket);
+            // Give LDAP client time to fully establish the connection.
+            // When client is done - close the accepted socket
+            testDone.await();
+        } catch (Exception e) {
+            System.err.println("Server socket. Failure to accept connection:");
+            e.printStackTrace();
         }
```

Comment:
I wonder if removing the while (true) loop will make the test more susceptible of failing in timeout if the server ever receives a connection request from an unexpected client (we've seen that happening in the past with networking tests). Is there anyway the server could attempt to verify that the accepted socket is from the expected client, and close it and go back to accepting if it's not? Maybe by looking at the accepted socket remote address & port?

Reply:
Thanks for the good suggestion Daniel. I will modify it to look at the remote socket's address and port.

Reply:
Hi Daniel,

Could you please take a look at the new version of DeadSSLLdapTimeoutTest.java with the following modifications: 
**DeadServerTimeoutSSLTest** test was modified to use custom **SocketFactory** (**DeadSSLSocketFactory**) which tracks the first opened socket on LDAP client side. This socket is later used on server side to check if the connection initiated by test's LDAP client. If the connection was not established by LDAP client then new accept attempt is performed.

The test was never seen to fail during 200+ concurrent LDAP tests runs.